### DISCUSSION TITLE
Multi-Cluster Communication between two EKS clusters

### DISCUSSION THREAD

**Discussion by sekar-saravanan** [istio/istio#54004] [General]
Hi Everyone,

I am setting up cross-cluster communication between two EKS clusters whose VPCs are already peered. With VPC peering, pods can communicate directly between the two clusters using pod IPs by default. However, service IPs are not accessible across clusters because the service CIDRs are internal to each cluster.

Initially, I attempted a Multi-Primary setup but was disappointed as it did not work in my scenario. Then, I tried the Multi-Primary on Different Networks setup using the East-West Gateway, which worked successfully.

Could you please clarify which multi-cluster installation method I should choose? Should I go with Multi-Primary or Multi-Primary on Different Networks? I came across this [suggestion](https://github.com/istio/istio/discussions/50723) to use Multi-Primary on Different Networks, but I want to ensure I understand the reasoning.

Additionally, I want to better understand the exact purpose of the East-West Gateway. Are there any documents or resources that explain its role? Specifically:

1. Is the East-West Gateway only for service discovery between clusters?
2. Does all inter-cluster traffic flow through the East-West Gateway?
3. If inter-cluster traffic indeed flows through the East-West Gateway, how can I avoid this? My clusters already have pod-to-pod communication through VPC peering, and I would like to eliminate the additional hop caused by routing traffic via the East-West Gateway’s load balancer.

**Comment by rootsongjc** [comment]
Based on your scenario, I’d recommend using the **Multi-Primary on Different Networks** setup. Since your clusters are on different service CIDRs and you already have pod-to-pod communication through VPC peering, this setup is better suited. It allows for service discovery across clusters using the East-West Gateway, solving the issue of inaccessible service IPs.

As for the East-West Gateway:
1. It's not just for service discovery—it also routes inter-cluster traffic securely.
2. By default, all inter-cluster traffic flows through it for routing and mTLS.
3. If you want to avoid the extra hop, you can create custom ServiceEntries or tweak DNS to leverage your VPC peering for direct pod-to-pod traffic. Just make sure security (like mTLS) is handled appropriately.

You might find this blog helpful: [A Definitive Guide to Cross-Cluster Seamless Access in Multicluster Istio Service Mesh](https://jimmysong.io/en/blog/seamless-cross-cluster-access-istio/#cross-cluster-service-registration-discovery-and-routing). It dives deeper into cross-cluster service discovery and routing in Istio.
